torch/optim/lr_scheduler.pyi

```diff
@@ -34,18 +51,61 @@ class SequentialLR(_LRScheduler):
     def __init__(self, optimizer: Optimizer, schedulers: List[_LRScheduler], milestones: List[int], last_epoch: int=..., verbose: bool=...) -> None: ...
 
 class CosineAnnealingLR(_LRScheduler):
-    def __init__(self, optimizer: Optimizer, T_max: int, eta_min: float=..., last_epoch: int=...) -> None: ...
+    T_max: int = ...
+    eta_min: float = ...
+    def __init__(self, optimizer: Optimizer, T_max: int, eta_min: float = ..., last_epoch: int = ..., verbose: bool = ...) -> None: ...
 
 class ReduceLROnPlateau:
-    in_cooldown: bool
-
-    def __init__(self, optimizer: Optimizer, mode: str=..., factor: float=..., patience: int=..., verbose: bool=..., threshold: float=..., threshold_mode: str=..., cooldown: int=..., min_lr: float=..., eps: float=...) -> None: ...
-    def step(self, metrics: Any, epoch: Optional[int]=...) -> None: ...
-    def state_dict(self) -> dict: ...
-    def load_state_dict(self, state_dict: dict): ...
+    factor: float = ...
+    optimizer: Optimizer = ...
+    min_lrs: List[float] = ...
+    patience: int = ...
+    verbose: bool = ...
+    cooldown: int = ...
+    cooldown_counter: int = ...
+    mode: str = ...
+    threshold: float = ...
+    threshold_mode: str = ...
+    best: Optional[float] = ...
+    num_bad_epochs: int = ...
+    mode_worse: Optional[float] = ...
+    eps: float = ...
+    last_epoch: int = ...
+    def __init__(self, optimizer: Optimizer, mode: str = ..., factor: float = ..., patience: int = ..., threshold: float = ..., threshold_mode: str = ..., cooldown: int = ..., min_lr: Union[List[float], float] = ..., eps: float = ..., verbose: bool = ...) -> None: ...
+    def step(self, metrics: Any, epoch: Optional[int] = ...) -> None: ...
+    @property
+    def in_cooldown(self) -> bool: ...
+    def is_better(self, a: Any, best: Any) -> bool: ...
 
 class CyclicLR(_LRScheduler):
-    def __init__(self, optimizer: Optimizer, base_lr: float=..., max_lr: float=..., step_size_up: int=..., step_size_down: int=..., mode: str=..., gamma: float=..., scale_fn: Optional[Callable[[float], float]]=..., scale_mode: str=..., cycle_momentum: bool=..., base_momentum: float=..., max_momentum: float=..., last_epoch: int=...) -> None: ...
+    optimizer: Optimizer = ...
+    max_lrs: Any = ...
+    total_size: Any = ...
+    step_ratio: Any = ...
+    mode: Any = ...
+    gamma: Any = ...
+    scale_fn: Any = ...
+    scale_mode: str = ...
+    cycle_momentum: Any = ...
+    base_momentums: Any = ...
+    max_momentums: Any = ...
+    base_lrs: Any = ...
```

Comment:
`base_lrs` is already defined in the base. Does this need to be here?

Reply:
No.